soufi/finders/java.py

```diff
@@ -22,21 +21,30 @@ def _find(self):
         source_url = self.get_source_url()
         return JavaDiscoveredSource([source_url])
 
+    # NOTE(nic): override this method to also return the response object so
+    #  that it may be inspected.
```

Comment:
It's a one-off, and not particularly useful/interesting to any other call sites, so not worth passing around elsewhere.

Making a multi-value return the default behavior just feels like a great way to generate `ValueError`s later.  And my preferred option (a bespoke object that evaluates truthiness normally, but also possesses properties) seems a little bit bonkers here.

Reply:
There's already a bespoke object — the `Response` ! Overriding here doesn't make any sense to me when you can do any of:
a) just return the Response everywhere.
b) raise an exception in the case the url is not found
c) return None in the case the URL is not found and always return the URL

I think it's best to just return the `Response` and raise an exception if it's not found. If you want the least work, return the Response or a None instead of True/False.

Sorry for nitpicking but this is going to be part of the public API on the base finder so it needs to be spot on.

Reply:
![nathan-fillion-meme](https://user-images.githubusercontent.com/1404587/152721114-59f93d38-179e-4b7f-bef2-9997fda1ef04.gif)

I don't really know what I was thinking before.